cirq-core/cirq/ops/state_preparation_channel.py

```diff
@@ -108,13 +107,21 @@ def _kraus_(self):
         return operator
 
     def __repr__(self) -> str:
-        return f'cirq.StatePreparationChannel({proper_repr(self._state)})'
+        return f'cirq.StatePreparationChannel({proper_repr(self.state)})'
```

Comment:
`name` should also be a part of the `repr`. Also add tests which verify this. 


Also, let's make `name` a keyword argument. It can be confusing to specify a string after the matrix without the argument name. 

So the init function will change to:

```python
def __init__(self, target_state: np.ndarray, *, name: str = "StatePreparation") -> None:
```

Reply:
Changed it, added name to `__init__`, `to_json`, `from_json`, `__repr__`, and JSON and repr test data.